plugins/offboard/offboard_impl.cpp

```diff
@@ -373,6 +400,55 @@ void OffboardImpl::send_velocity_body()
     _parent->send_message(message);
 }
 
+void OffboardImpl::send_attitude()
+{
+    const static uint8_t IGNORE_BODY_ROLL_RATE = (1 << 0);
+    const static uint8_t IGNORE_BODY_PITCH_RATE = (1 << 1);
+    const static uint8_t IGNORE_BODY_YAW_RATE = (1 << 2);
+    // const static uint8_t IGNORE_4 = (1 << 3);
+    // const static uint8_t IGNORE_5 = (1 << 4);
+    // const static uint8_t IGNORE_6 = (1 << 5);
+    // const static uint8_t IGNORE_THRUST = (1 << 6);
+    // const static uint8_t IGNORE_ATTITUDE = (1 << 7);
+
+    _mutex.lock();
+    const float thrust = _attitude.thrust_value;
+    const float roll = to_rad_from_deg(_attitude.roll_deg);
+    const float pitch = to_rad_from_deg(_attitude.pitch_deg);
+    const float yaw = to_rad_from_deg(_attitude.yaw_deg);
+    _mutex.unlock();
+
+    const double cos_phi_2 = cos(double(roll) / 2.0);
+    const double sin_phi_2 = sin(double(roll) / 2.0);
+    const double cos_theta_2 = cos(double(pitch) / 2.0);
+    const double sin_theta_2 = sin(double(pitch) / 2.0);
+    const double cos_psi_2 = cos(double(yaw) / 2.0);
+    const double sin_psi_2 = sin(double(yaw) / 2.0);
+
+    float q[4];
+
+    q[0] = float(cos_phi_2 * cos_theta_2 * cos_psi_2 + sin_phi_2 * sin_theta_2 * sin_psi_2);
+    q[1] = float(sin_phi_2 * cos_theta_2 * cos_psi_2 - cos_phi_2 * sin_theta_2 * sin_psi_2);
+    q[2] = float(cos_phi_2 * sin_theta_2 * cos_psi_2 + sin_phi_2 * cos_theta_2 * sin_psi_2);
+    q[3] = float(cos_phi_2 * cos_theta_2 * sin_psi_2 - sin_phi_2 * sin_theta_2 * cos_psi_2);
```

Comment:
I think you can re-use this:
https://github.com/Dronecode/DronecodeSDK/blob/4f591ad830f3cf7bd388418dcebef14a4536aeca/plugins/telemetry/math_conversions.cpp#L22-L38

Reply:
That was indeed the source of inspiration. Isn't some restructuring necessary? Seems odd that one plugin is dependant of internal files within another plugin?

Reply:
Oh you're right! Let's leave it as is for now and clean it up later.